state/protocol/badger/validity.go

```diff
@@ -3,16 +3,18 @@ package badger
 import (
 	"fmt"
 
-	"github.com/onflow/flow-go-sdk/crypto"
+	"github.com/onflow/flow-go/crypto"
 	"github.com/onflow/flow-go/model/flow"
 	"github.com/onflow/flow-go/model/flow/filter"
+	"github.com/onflow/flow-go/state/protocol"
 )
 
 func validSetup(setup *flow.EpochSetup) error {
 	// STEP 1: general sanity checks
 	// the seed needs to be at least minimum length
-	if len(setup.RandomSource) < crypto.MinSeedLength {
-		return fmt.Errorf("seed has insufficient length (%d < %d)", len(setup.RandomSource), crypto.MinSeedLength)
+	// TODO: what is the appropriate minimum length here? Previously used sdk/crypto.SeedMinLength
```

Comment:
@tarakby Maybe you can answer this. This `RandomSource` is the random source used to generate random seeds with the `Snapshot.Seed` API.

Reply:
Do you know do we generate `setup.RandomSource` ? is it by using the package `crypto/rand` or is it a human chosen passphrase?

Reply:
It's ultimately generated by the random beacon, so I think we want the size of the random beacon output here (also cc @AlexHentschel)

Reply:
ah okay, in that case it has to be exactly `crypto.SignatureLenBLSBLS12381`.
 
It might make sense to have another constant `RandomSourceLength` that the protocol uses and it is rooted to the BLS length in `crypto.`

Reply:
Not sure how deep we want `validSetup(setup *flow.EpochSetup)` to validate, but checking just the length doesn't give much guarantee about the random source. 
Maybe we could cast `setup.RandomSource` as a signature and verify it against the random beacon public key of the current epoch, if it's part of the snapshot.

Reply:
Updated the min length in https://github.com/onflow/flow-go/pull/312/commits/0130321f25d89909a8b07a6c6e1cc5fffe66b67b

> Maybe we could cast setup.RandomSource as a signature and verify it against the random beacon public key of the current epoch

The idea is `RandomSource` will eventually come from an updated Cadence random API -- might make sense to have this check once that is in place depending on the details. For now I will leave only the length sanity check.